api/apps/api/src/modules/access-control/scenarios-acl/dto/user-role-scenario.dto.ts

```diff
@@ -0,0 +1,29 @@
+import { Roles } from '@marxan-api/modules/access-control/role.api.entity';
+import { ApiProperty } from '@nestjs/swagger';
+import { IsEnum, IsString, IsUUID } from 'class-validator';
+
+export class UserRoleInScenarioDto {
+  @IsUUID()
+  @ApiProperty()
+  userId!: string;
+
+  @IsUUID()
+  @ApiProperty()
+  scenarioId!: string;
+
+  @IsString()
+  @IsEnum(Object.values(Roles))
+  @ApiProperty()
+  roleName?:
+    | Roles.scenario_viewer
+    | Roles.scenario_contributor
+    | Roles.scenario_owner;
```

Comment:
1 This will allow to bypass the validation if different role is assigned
2 `ApiProperty` may need more details - otherwise it may not render in swagger.